docs/dev/setup/programming-exercises.rst

```diff
@@ -60,3 +60,30 @@ Note that the ``Jenkinsfile`` **must** start either
 The variables `#dockerImage`, `#testRepository`, `#assignmentRepository`, `#jenkinsNotificationToken` and `#notificationsUrl` will automatically be replaced (for the normal Jenkinsfile, within the Jenkinsfile-staticCodeAnalysis, #staticCodeAnalysisScript is also replaced).
 
 You should not need to touch any of these variables, except the `#dockerImage` variable, if you want to use a different agent setup (e.g. a Kubernetes setup).
+
+
+Caching example for Maven
+^^^^^^^^^^^^^^^^^^^^^^^^^
+Adjust the agent-args and add the environment block.
```

Comment:
Can you please add a reason why this is necessary. I guess the main reason is to speed up build times...

Reply:
Yes. It significantly speeds up build. Sady the dependencies cached in the docker image are frequently out of date. So we do the caching like that

Reply:
I've added an explanation.